src/tests/mock_observer.hpp

```diff
@@ -20,7 +20,8 @@ template<typename Type>
 class mock_observer : public rpp::interface_observer<Type>
 {
 public:
-    mock_observer(bool copy_values = true) : m_state{std::make_shared<State>(copy_values)} {}
+    explicit mock_observer(bool copy_values = true) : m_state{std::make_shared<State>(copy_values)} {}
+    ~mock_observer() noexcept = default;
```

Comment:
Do we really need it? Anyway, it must have `override`

Reply:
No, we don't really need the dtor.

Reply:
Addressed in the latest commit.